### PR TITLE
Fixes Release notes issue.

### PR DESCRIPTION
**Pull request by raujaiswal**
### **Context**
Additional fixes applied over the previous PR: https://github.com/microsoft/azure-pipelines-agent/pull/5182

### **Description**
Generating release notes for the latest release version by comparing the lastMinorRelease version with the master branch. Additional pull requests (PRs) merged since the last release are included, considering only the PRs up to the specified commit ID.

**How to check/verify the value of Build.SourceVersion, follow these steps:**

**Step 1:**
Note that Build.SourceVersion is already used in the YAML pipeline for creating the GitHub release. You can see its usage by searching for target_commitish in the repository:
[GitHub Search: target_commitish in azure-pipelines-agent](https://github.com/search?q=repo%3Amicrosoft%2Fazure-pipelines-agent%20target_commitish&type=code)

**Step 2:**
Go to a previously successful pipeline run, for example:
[Azure DevOps Pipeline Logs](https://mseng.visualstudio.com/AzureDevOps/_build/results?buildId=29932112&view=logs&j=2d906efd-181e-5a2c-ecb8-0d2e66617d51&t=4b948b09-3ca3-5ca2-27ca-f7a233efb27f)
Search for **target_commitish** in the logs to find its value.

**Step 3:**
Verify that the value of target_commitish matches the commit ID present in the [GitHub commits for master branch](https://github.com/microsoft/azure-pipelines-agent/commits/master/).

---

### **Risk Assessment** (Low / Medium / High)  
Medium

---

### **Unit Tests Added or Updated** (Yes / No)  
No

---

### **Additional Testing Performed**
Yes, test performed locally.

### CODE REVIEW

File: .vsts.release.yml [hunk 1]
```diff
@@ -194,7 +194,7 @@ extends:
             cd release
             npm install
 
-            node createReleaseBranch.js $(AgentVersion) --derivedFrom=${{ parameters.derivedFrom }}
+            node createReleaseBranch.js $(AgentVersion) --derivedFrom=${{ parameters.derivedFrom }} --targetCommitId=$(Build.SourceVersion)
```
Comment (sanjuyadav24) [inline .vsts.release.yml:197]: is user has not added any specific commit, this will pick the latest commit from master?

Comment (sanjuyadav24) [reply .vsts.release.yml:197]: lets make sure we update the release process documentation accordingly to incorporate this change

Comment (tarunramsinghani) [reply .vsts.release.yml:197]: +1, did you test it with empty commit mentioned in the pipeline.

Comment (raujaiswal) [reply .vsts.release.yml:197]: By default, if no specific commit is provided, it will pick the latest commit from the master branch. I also tested it with an empty commit; in such cases, it will include all the changes based on the comparison between the master branch and the lastMinorRelease.

Comment (raujaiswal) [reply .vsts.release.yml:197]: To check the value of Build.SourceVersion, follow these steps:

Step 1:
Note that Build.SourceVersion is already used in the YAML pipeline for creating the GitHub release. You can see its usage by searching for target_commitish in the repository:
[GitHub Search: target_commitish in azure-pipelines-agent](vscode-file://vscode-app/c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)

Step 2:
Go to a previously successful pipeline run, for example:
[Azure DevOps Pipeline Logs](vscode-file://vscode-app/c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)
Search for target_commitish in the logs to find its value.

Step 3:
Verify that the value of target_commitish matches the commit ID present in the [GitHub commits for master branch](vscode-file://vscode-app/c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).

This process ensures that Build.SourceVersion is correctly set and corresponds to the intended commit in the repositor

Comment (raujaiswal) [reply .vsts.release.yml:197]: Updated PR Description: How to Cross-Check the Value of Build.SourceVersion

Comment (sanjuyadav24) [reply .vsts.release.yml:197]: I think we should update the agent release documentation for its adoption
as currently commit id is an optional field, we are not selecting this while running pipeline
Also it would be great if a check can be added that commit id field is not empty

Comment (sanjuyadav24) [reply .vsts.release.yml:197]: we should also define the rules which commit should be picked for release, eg PRs merged till this date will go in the release

Comment (raujaiswal) [reply .vsts.release.yml:197]: Discussed on the call: if we make the commit ID mandatory, it could create overhead since we’d have to manually find and enter the commit ID each time. In most cases, we actually want to use the latest commit ID by default.

**Review by sanjuyadav24 — APPROVED** [review]

**Review by tarunramsinghani — APPROVED** [review]
